### PR TITLE
Updates widget unit tests to use the latest test-extras

### PR DESCRIPTION
**Pull request by agubler**
**Type:** feature

The following has been addressed in the PR:

* [ ] There is a related issue
* [x] All code matches the [style guide](https://github.com/dojo/meta/blob/master/STYLE.md)
* [x] Unit or Functional tests are included in the PR

**Description:**

Sorry about the large changes :) Updates all the widget unit tests to use the upcoming latest `@dojo/test-extras` release with a brand new API.

Introduces a new `GlobalEvent` widget that can be used to register events on `document` and `window`.

There were only a few tests that we were unable to implement with the updated release as no widgets are attached to the DOM by `@dojo/test-extras`.

### CODE REVIEW

File: src/calendar/tests/unit/DatePicker.ts [hunk 1]
```diff
@@ -194,296 +196,220 @@ interface TestEventInit extends EventInit {
...
-			});
-			expectedVdom = expected(widget, false, false, 1980, 2000);
-			const yearGridVdom = findKey(expectedVdom, 'year-grid');
-			replaceChild(expectedVdom, '0,0,0', 'June 1997');
-			replaceChild(expectedVdom, '0,2,0', '1997');
-			assignChildProperties(yearGridVdom!, '0,18', { classes: [ css.yearRadio, css.yearRadioChecked ] });
-			assignChildProperties(yearGridVdom!, '0,18,0', { checked: true });
-			widget.expectRender(expectedVdom);
-		},
+		// 'Year below 2000 calculates correctly'() {
```
Comment (agubler) [inline src/calendar/tests/unit/DatePicker.ts]: Need to add this test still

**Comment by tomdye** [comment]
@agubler nice work, that was a lot of changes!

**Comment by tomdye** [comment]
Can you create an issue to update the readme re. `GlobalEvent` perhaps?

**Comment by agubler** [comment]
@tomdye added a README for `GlobalEvent`

**Pull request merged**